// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: service/models.proto

package service

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
)

// Validate checks the field values on ProdModel with the rules defined in the
// proto definition for this message. If any rules are violated, an error is returned.
func (m *ProdModel) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for ProdId

	// no validation rules for ProdName

	// no validation rules for ProdPrice

	return nil
}

// ProdModelValidationError is the validation error returned by
// ProdModel.Validate if the designated constraints aren't met.
type ProdModelValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ProdModelValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ProdModelValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ProdModelValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ProdModelValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ProdModelValidationError) ErrorName() string { return "ProdModelValidationError" }

// Error satisfies the builtin error interface
func (e ProdModelValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sProdModel.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ProdModelValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ProdModelValidationError{}

// Validate checks the field values on OrderMain with the rules defined in the
// proto definition for this message. If any rules are violated, an error is returned.
func (m *OrderMain) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for OrderId

	// no validation rules for OrderNo

	// no validation rules for UserId

	if m.GetOrderMoney() <= 99 {
		return OrderMainValidationError{
			field:  "OrderMoney",
			reason: "value must be greater than 99",
		}
	}

	if v, ok := interface{}(m.GetOrderTime()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return OrderMainValidationError{
				field:  "OrderTime",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	for idx, item := range m.GetDetails() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return OrderMainValidationError{
					field:  fmt.Sprintf("Details[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// OrderMainValidationError is the validation error returned by
// OrderMain.Validate if the designated constraints aren't met.
type OrderMainValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderMainValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderMainValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderMainValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderMainValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderMainValidationError) ErrorName() string { return "OrderMainValidationError" }

// Error satisfies the builtin error interface
func (e OrderMainValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrderMain.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderMainValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderMainValidationError{}

// Validate checks the field values on OrderDetail with the rules defined in
// the proto definition for this message. If any rules are violated, an error
// is returned.
func (m *OrderDetail) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for OrderNo

	// no validation rules for DetailId

	return nil
}

// OrderDetailValidationError is the validation error returned by
// OrderDetail.Validate if the designated constraints aren't met.
type OrderDetailValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderDetailValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderDetailValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderDetailValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderDetailValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderDetailValidationError) ErrorName() string { return "OrderDetailValidationError" }

// Error satisfies the builtin error interface
func (e OrderDetailValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrderDetail.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderDetailValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderDetailValidationError{}

// Validate checks the field values on UserScore with the rules defined in the
// proto definition for this message. If any rules are violated, an error is returned.
func (m *UserScore) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for UserId

	// no validation rules for Score

	return nil
}

// UserScoreValidationError is the validation error returned by
// UserScore.Validate if the designated constraints aren't met.
type UserScoreValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UserScoreValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UserScoreValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UserScoreValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UserScoreValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UserScoreValidationError) ErrorName() string { return "UserScoreValidationError" }

// Error satisfies the builtin error interface
func (e UserScoreValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUserScore.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UserScoreValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UserScoreValidationError{}
